src/main/java/fr/insalyon/creatis/gasw/execution/GaswSubmit.java

```diff
@@ -85,46 +93,65 @@ public GaswSubmit(GaswInput gaswInput, GaswMinorStatusServiceGenerator minorStat
      *
      * @return
      * @throws GaswException
+     * @throws IOException 
      */
-    protected String generateScript() throws GaswException {
-
-        String script = ScriptGenerator.getInstance().generateScript(
-                gaswInput, minorStatusServiceGenerator);
-
-        return publishScript(gaswInput.getExecutableName(), script);
-    }
-
-    /**
-     *
-     * @param symbolicName Symbolic name of the execution.
-     * @param script Generated script to be saved in a file.
-     * @return Name of the script file.
-     */
-    private String publishScript(String symbolicName, String script) {
-
-        try {
-            File scriptsDir = new File(GaswConstants.SCRIPT_ROOT);
-            if (!scriptsDir.exists()) {
-                scriptsDir.mkdir();
+    protected String generateScript() throws GaswException, IOException {
+        String scriptName;
+
+            if (gaswInput.isMoteurLiteEnabled()) {
+                // Logic for Moteurlite-specific script generation
+                logger.info("MoteurLite is enabled, generating Moteurlite-specific script.");
+                
+                // Get the script from resources
+                String scriptMoteurlite = readScriptFromResources();
+                
+                // Generate the Moteurlite-specific configuration
+                Map<String, String> configMoteurlite = MoteurliteConfigGenerator.getInstance().generateConfig(gaswInput, minorStatusServiceGenerator);
+                
+                // Publish the configuration and invocation
+                publishConfiguration(gaswInput.getJobId(), configMoteurlite);
+                publishInvocation(gaswInput.getJobId(), gaswInput.getInvocationString());
+                
+                // Publish the script itself
+                scriptName = publishScript(gaswInput.getExecutableName(), scriptMoteurlite);
+
+            } else {
+                // Original default behavior (when Moteurlite is not enabled)
+                String defaultScript = ScriptGenerator.getInstance().generateScript(gaswInput, minorStatusServiceGenerator);
+                scriptName = publishScript(gaswInput.getExecutableName(), defaultScript);
             }
+  
+        return scriptName;
+    }
 
-            String fileName = symbolicName.replace(" ", "-");
+    private String publishScript(String symbolicName, String script) throws IOException {
+        String fileName = null;
+    
+        // Ensure the script directory exists
+        File scriptsDir = new File(GaswConstants.SCRIPT_ROOT);
+        if (!scriptsDir.exists()) {
+            scriptsDir.mkdir();
+        }
+    
+        // If MoteurLite is enabled, use the jobId as the script name
+        if (gaswInput.isMoteurLiteEnabled()) {
+            fileName = gaswInput.getJobId(); 
+            writeToFile(GaswConstants.SCRIPT_ROOT + "/" + fileName, script);
```

Comment:
Could it be possible to not read the script as string and then rewrite it, and instead to just copy the file ?